stdlib/websub/src/main/ballerina/src/websub/hub_configuration.bal

```diff
@@ -36,6 +36,9 @@ RemotePublishConfig remotePublishConfig = {};
 boolean hubTopicRegistrationRequired = false;
 string hubPublicUrl = "";
 http:ClientConfiguration? hubClientConfig = ();
+function (WebSubContent content)? onMessageFunction = ();
+function (string callback, string topic, WebSubContent content)? onDeliveryFunction = ();
+function (string callback, string topic, WebSubContent content, http:Response|error response)? onDeliveryFailureFunction = ();
```

Comment:
Do we need to execute the current logic (logging) if the user has specified a custom function? 

Instead of making this nilable shall we just have the functions and assign as the default implementation the current logic that logs?

Given that delivery failure can happen due to three reasons, we can add maybe add another `FailureReason` parameter to quickly figure out what went wrong, and in the default case log the appropriate log.


```ballerina
const FAILURE_REASON_SUBSCRIPTION_GONE = "SUBSCRIPTION_GONE";
const FAILURE_REASON_FAILURE_STATUS_CODE = "FAILURE_STATUS_CODE";
const FAILURE_REASON_DELIVERY_FAILURE = "DELIVERY_FAILURE";

type FailureReason FAILURE_REASON_SUBSCRIPTION_GONE|FAILURE_REASON_FAILURE_STATUS_CODE|FAILURE_REASON_DELIVERY_FAILURE;

function (string callback, string topic, WebSubContent content, http:Response|error response, FailureReason reason) onDeliveryFailureFunction = defaultOnDeliveryFailureFunction;
```

This way we can avoid having to use the type test per delivery.

Reply:
The `onMessageFunction` default implementation can be an empty function or a debug log.

Reply:
+1. Changed the code as requested.

Reply:
I think in all the other cases, we simply tap the flow. In other words, the core functionality stays the same regardless of what we are doing. Doing what is suggested make this function in consistent with the other two if I am not mistaken.